theme-settings.php

```
@@ -117,6 +106,244 @@ function dxpr_theme_form_system_theme_settings_alter(&$form, &$form_state, $form
+            setTimeout(function() {
+              indexSearchableElements();
+            }, 500);
+            
+            // Re-index when new content is loaded (for dynamic content)
+            var observer = new MutationObserver(function() {
+              clearTimeout(searchTimeout);
+              searchTimeout = setTimeout(function() {
+                indexSearchableElements();
+              }, 300);
+            });
+            observer.observe(themeSettings, { childList: true, subtree: true });
+            
+            // Also re-index on window load and when vertical tabs are clicked
+            window.addEventListener('load', function() {
+              setTimeout(indexSearchableElements, 1000);
+            });
+            
+            // Listen for vertical tab clicks to re-index
+            document.addEventListener('click', function(e) {
+              if (e.target.closest('.vertical-tabs__menu-item')) {
+                setTimeout(indexSearchableElements, 100);
+              }
+            });
+          }
+          
+          requestAnimationFrame(function() {
+            var contentRegion = document.querySelector('.region-content');
+            if (contentRegion) {
+              var styleguideDiv = document.createElement('div');
+              styleguideDiv.innerHTML = '<h2>Bootstrap Styleguide</h2><p>Loading...</p>';
+              contentRegion.insertBefore(styleguideDiv, contentRegion.firstChild);
+              
+              fetch('" . $styleguide_url . "')
+                .then(function(response) { return response.text(); })
+                .then(function(html) {
+                  var parser = new DOMParser();
+                  var doc = parser.parseFromString(html, 'text/html');
+                  var cheatsheet = doc.querySelector('.bd-cheatsheet');
+                  if (cheatsheet) {
+                    styleguideDiv.innerHTML = '<h2>Bootstrap Styleguide</h2>' + cheatsheet.outerHTML;
```

} else {
      styleguideDiv.innerHTML = '<h2>Bootstrap Styleguide</h2><p>Could not find cheatsheet content.</p>';
    }
  })
  .catch(function(err) {
    styleguideDiv.innerHTML = '<h2>Bootstrap Styleguide</h2><p>Failed to load styleguide (' + err.message + ').</p>';
  });
  
  Add a user-facing fallback so admins see a clear message if the styleguide fails to load or the expected .bd-cheatsheet isn’t present, instead of silent failure. This improves UX and diagnostics without affecting performance.